player/cmcd/js/script.js

```diff
@@ -0,0 +1,268 @@
+const LOG_PULL_INTERVALL_SECONDS = 10 * 1000;
+
+const FIELD = {
+  TIMESTAMP: 4,
+  HTTP_STATUS: 6,
+  HTTP_VERSION: 7,
+  HOST: 8,
+  METHOD: 9,
+  PATH: 10,
+  PORT: 11,
+  MIME_TYPE: 14,
+  USER_AGENT: 15,
+  ENCRYPTION_VERSION: 16,
+  CMCD_DATA: 36,
+};
+const DELIMITER = '&nbsp;'
+const CMCD_KEYS = {
+  EncodedBitrate: 'br',
+  BufferLength: 'bl',
+  BufferStarvation: 'bs',
+  ContentId: 'cid',
+  ObjectDuration: 'd',
+  Deadline: 'dl',
+  MeasuredThroughput: 'mtp',
+  NextObjectRequest: 'nor',
+  NextRangeRequest: 'nrr',
+  ObjectType: 'ot',
+  PlaybackRate: 'pr',
+  RequestedMaximumThroughput: 'rtp',
+  StreamingFormat: 'sf',
+  SessionId: 'sid',
+  StreamType: 'st',
+  Startup: 'su',
+  TopBitrate: 'tb',
+  CmcdVersion: 'v',
+}
+
+const logsToDisplay = [];
+let lastSuccessfulRetrieval = new Date();
+let cmcdSessionId;
+
+function setupPlayerWithCmcd() {
+  cmcdSessionId = uuidv4();
+
+  const playerConfig = {
+    key: '29ba4a30-8b5e-4336-a7dd-c94ff3b25f30',
+    analytics: {
+      key: '45adcf9b-8f7c-4e28-91c5-50ba3d442cd4',
+      videoId: 'cmcd'
+    },
+    playback: {
+      muted: true,
+      autoplay: true,
+    }
+  };
+
+  const cmcdConfig = {
+    sessionId: cmcdSessionId,
+    contentId: '1111-111111-111111-11111',
+  };
+
+  const cmcdPlugin = new window.bitmovin.player.integration.Cmcd.CmcdIntegration(cmcdConfig);
+  playerConfig.network = {
+    preprocessHttpRequest: cmcdPlugin.preprocessHttpRequest,
+    preprocessHttpResponse: cmcdPlugin.preprocessHttpResponse,
+  };
+  playerConfig.adaptation = {
+    desktop: {
+      onVideoAdaptation: cmcdPlugin.onVideoAdaptation,
+      onAudioAdaptation: cmcdPlugin.onAudioAdaptation,
+    },
+    mobile: {
+      onVideoAdaptation: cmcdPlugin.onVideoAdaptation,
+      onAudioAdaptation: cmcdPlugin.onAudioAdaptation,
+    },
+  };
```

Comment:
```suggestion
  const cmcdConfig = {
    sessionId: cmcdSessionId,
    contentId: '1111-111111-111111-11111',
  };
  const cmcdPlugin = new window.bitmovin.player.integration.Cmcd.CmcdIntegration(cmcdConfig);
  
  const playerConfig = {
    key: '29ba4a30-8b5e-4336-a7dd-c94ff3b25f30',
    analytics: {
      key: '45adcf9b-8f7c-4e28-91c5-50ba3d442cd4',
      videoId: 'cmcd'
    },
    playback: {
      muted: true,
      autoplay: true,
    },
    network: {
      preprocessHttpRequest: cmcdPlugin.preprocessHttpRequest,
      preprocessHttpResponse: cmcdPlugin.preprocessHttpResponse,
    },
    adaptation: {
      onVideoAdaptation: cmcdPlugin.onVideoAdaptation,
      onAudioAdaptation: cmcdPlugin.onAudioAdaptation,
    }
  };
```
I think it would be more readable when all the config is in one place. Also I think (please double check) that we don't need video and desktop entries for the adaptation.

Reply:
> Also I think (please double check) that we don't need video and desktop entries for the adaptation.
I thought the same and I was sure I did test it when I implemented this and it didn't work for me. Testing again it works 🤷 Thanks 😅 

Fair point about all config in one place. I did take this from the CMCD integration project's README as this easily works with any player config (as long as the adaptation and network callbacks aren't already used). For the demo it should be fine either way.